src/components/WalletButton/WalletButton.tsx

```diff
@@ -29,37 +47,111 @@ export type WalletButtonProps = {
-            seed={address}
-            className="-ml-1 rounded-sm mr-2"
-            bgColor="black"
-            color="#2b72ff"
-          />
-          <span>{truncateEthAddress(address)}</span>
-          <button
-            className="ml-2"
-            aria-label={t("wallet:disconnectWallet")}
-            onClick={onDisconnectWalletClicked}
-          >
-            <RiCloseLine />
-          </button>
-        </StyledWalletButton>
+        <>
+          <StyledWalletButton onClick={() => setWalletOpen(!walletOpen)}>
+            <BlockiesContainer>
+              <StyledBlockies
+                size={8}
+                scale={3}
+                seed={address}
+                bgColor="black"
+                color="#2b72ff"
+              />
+              <GreenCircle />
+            </BlockiesContainer>
+            <InfoHeading>{truncateEthAddress(address)}</InfoHeading>
+            {walletOpen && (
+              <ExitButton onClick={() => setWalletOpen(!walletOpen)}>
+                <RiCloseLine />
+              </ExitButton>
+            )}
+          </StyledWalletButton>
+          {walletOpen && (
+            <WalletExtension>
+              <Line />
+              <TransactionContainer flex={transactions.length === 0}>
+                {transactions.length > 0 ? (
+                  transactions.slice(0, 3).map((transaction) => {
+                    let token;
+                    if (transaction.type === "Order") {
+                      const tx: SubmittedOrder = transaction as SubmittedOrder;
+                      const senderToken = findTokenByAddress(
+                        tx.order.senderToken,
+                        tokens
+                      );
+                      const signerToken = findTokenByAddress(
+                        tx.order.signerToken,
+                        tokens
+                      );
+                      return (
+                        <TransactionRow
+                          transaction={transaction}
+                          senderToken={senderToken}
+                          signerToken={signerToken}
+                          type={transaction.type}
+                          chainId={chainId!}
+                          key={transaction.hash}
+                        />
+                      );
+                    } else if (transaction.type === "Approval") {
+                      const tx: SubmittedApproval = transaction as SubmittedApproval;
+                      token = findTokenByAddress(tx.tokenAddress, tokens);
+                      return (
+                        <TransactionRow
+                          transaction={transaction}
+                          approvalToken={token}
+                          type={transaction.type}
+                          chainId={chainId!}
+                          key={transaction.hash}
+                        />
+                      );
+                    }
+                    return null;
```

Comment:
There's too much logic in your render. I would make a useEffect with transactions as dependency that generates an array of transactions based of TransactionRowProps. This way your render is way cleaner.

Also what's the return null at the end for? :p